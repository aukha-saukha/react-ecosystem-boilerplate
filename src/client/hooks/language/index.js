// @flow strict

import { LanguageProvider, useLanguage } from './language.react';

export { LanguageProvider, useLanguage };
