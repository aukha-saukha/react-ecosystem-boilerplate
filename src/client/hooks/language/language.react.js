// @flow strict

import * as React from 'react';

import { DEFAULT_LOCALE } from '../../../data/constants/app/config';

import getUserLocale from '../../utilities/user-locale';

import type { LanguageProviderParametersType } from './language.type';

// TODO: Fix flow error here.
const LanguageContext: React.Context<
  [string, (((string) => string) | string) => void]
  // flow-disable-line
> = React.createContext();

LanguageContext.displayName = 'LanguageContext';

function LanguageProvider(props: LanguageProviderParametersType) {
  const { children } = props;
  const [language, setLanguage] = React.useState<string>(DEFAULT_LOCALE);

  React.useEffect(() => {
    async function getUserLocaleFromLocaleUtility() {
      const userLanguage = await getUserLocale();
      setLanguage(userLanguage);
    }

    getUserLocaleFromLocaleUtility();
  }, [language]);

  const providerValue = React.useMemo(() => [language, setLanguage], [language]);
  return <LanguageContext.Provider value={providerValue}>{children}</LanguageContext.Provider>;
}

function useLanguage() {
  const context = React.useContext(LanguageContext);

  if (typeof context === 'undefined') {
    throw new Error(`useLanguage must be used within a LanguageProvider`);
  }

  const [language, setLanguage] = context;

  return {
    language,
    setLanguage,
  };
}

export { LanguageProvider, useLanguage };
