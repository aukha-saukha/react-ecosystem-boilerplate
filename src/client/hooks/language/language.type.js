// @flow strict

import type { Node } from 'react';

export type LanguageProviderParametersType = {|
  children: Node,
|};
