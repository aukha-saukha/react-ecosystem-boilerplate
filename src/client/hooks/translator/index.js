// @flow strict

import useTranslator from './translator.react';

export default useTranslator;
