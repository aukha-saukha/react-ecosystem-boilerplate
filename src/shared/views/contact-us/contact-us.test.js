import { shallow } from 'enzyme';
import React from 'react';

import ContactUs from './contact-us.react';

import { LanguageProvider } from '../../../client/hooks/language';

describe('ContactUs component', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(
      <LanguageProvider>
        <ContactUs />
      </LanguageProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
