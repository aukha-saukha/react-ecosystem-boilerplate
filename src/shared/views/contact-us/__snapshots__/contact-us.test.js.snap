// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs component Renders correctly 1`] = `
<ContextProvider
  value={
    Array [
      "en-us",
      [Function],
    ]
  }
>
  <ContactUs />
</ContextProvider>
`;
